GameOfRunes/GameOfRunes/Levels and Waves/EnemySpawnUnit.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  EnemySpawnUnit.swift
+//  GameOfRunes
+//
+//  Created by Brian Yen on 28/3/20.
+//  Copyright © 2020 TeamHoWan. All rights reserved.
+//
+
+/**
+ This struct is used as the basic unit for creating a level. Different instances
+ of it can be concatenated together to create different types of levels. See
+ `LevelCreator` for more examples on how `EnemySpawnUnit` can
+  be used to create interesting levels.
+
+ - Note: The position of the `EnemyType` within the spawn wave (i.e. the [EnemyType]
+ in `unit` will determine the spawn position of the enemy (which lane the
+ enemy will be spawned at).
+ */
+struct EnemySpawnUnit {
+    var unit: [[EnemyType]]
```

Comment:
Feel that you shouldn't use an enum for this and just store it as a number, so that you won't have the issue of having a weak reference when there is no enemy

Reply:
Easier to persist into db also, and just do the conversion to enemy type when you actually do the spawning

Reply:
MM yup i see your point. I thought of that but decided against it because I felt that the level is determined more by the actual monster itself rather than the difficulty. So if we change the difficulty in the gameconfig, then it's possible for the stored level to become invalid. 

But then again, if somehow for multiplayer we need to be able to sync this data, then sending numbers would be more efficient and a whole lot easier. 

Maybe we KIV this for now?

Reply:
🆗